src/int/sign.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{ConstChoice, ConstCtOption, Int, Uint};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// Construct new [`Int`] from a sign and magnitude.
+    /// Returns `None` when the magnitude does not fit in an [`Int<LIMBS>`].
+    pub const fn new_from_abs_sign(
+        abs: Uint<LIMBS>,
+        is_negative: ConstChoice,
+    ) -> ConstCtOption<Self> {
+        let magnitude = Self(abs).wrapping_neg_if(is_negative);
+        let fits = Uint::gt(&abs, &Int::MAX.0)
+            .not()
+            .or(is_negative.and(Uint::eq(&abs, &Int::MIN.0)));
+        ConstCtOption::new(magnitude, fits)
```

Comment:
The current implementation uses two full array iterations: one for `gt` and one for `eq`. I believe this can be optimized to only be one iteration:

```suggestion
        let magnitude = Self(abs);
        let magnitude_is_negative = magnitude.is_negative();
        let value = magnitude.wrapping_neg_if(is_negative);
        let is_min = magnitude_is_negative.and(is_negative).and(value.is_negative())
        let fits = Uint::lte(&abs, &Int::MAX.0).or(is_min);
        ConstCtOption::new(value, fits)
```

hypothetically, this should be faster. we should benchmark to make sure though

Reply:
(also, I snuck in the `lte` already here :see_no_evil: )

Reply:
Fair. I just want to close the gap on the `Uint` and `Int` operations ;-)

Reply:
I'll close this for now; perhaps something to look into in the future.